### ISSUE TITLE
some pod stuck on ContainerCreating

### ISSUE THREAD

**Issue by H-W-WXL** [kubernetes/kubernetes#109665]
### What happened?

Using calico as a network plugin,Creating other pods before the calico node is successfully started,Even if calico returns to normal,other pods might  stuck on ContainerCreating and K8s does not retry that.
According to the kubelet log, the following error is reported when the kubelet is running:
`Apr 25 14:59:14 node0 kubelet[2447606]: E0425 14:59:14.836105 2447606 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
Apr 25 14:59:14 node0 kubelet[2447606]: goroutine 17927 [running]:
Apr 25 14:59:14 node0 kubelet[2447606]: k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime.logPanic(0x558b455bd820, 0x558b482ab440)
Apr 25 14:59:14 node0 kubelet[2447606]:         /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0x95
Apr 25 14:59:14 node0 kubelet[2447606]: k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
Apr 25 14:59:14 node0 kubelet[2447606]:         /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x87
Apr 25 14:59:14 node0 kubelet[2447606]: panic(0x558b455bd820, 0x558b482ab440)
Apr 25 14:59:14 node0 kubelet[2447606]:         /usr/local/go/src/runtime/panic.go:965 +0x1b9
Apr 25 14:59:14 node0 kubelet[2447606]: k8s.io/kubernetes/pkg/kubelet/kuberuntime.(*kubeGenericRuntimeManager).podSandboxChanged(0xc00084c180, 0xc001504800, 0xc00099d800, 0x0, 0x0, 0x0)
Apr 25 14:59:14 node0 kubelet[2447606]:         /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/kuberuntime/kuberuntime_manager.go:512 +0x140
Apr 25 14:59:14 node0 kubelet[2447606]: k8s.io/kubernetes/pkg/kubelet/kuberuntime.(*kubeGenericRuntimeManager).computePodActions(0xc00084c180, 0xc001504800, 0xc00099d800, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
Apr 25 14:59:14 node0 kubelet[2447606]:         /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/kuberuntime/kuberuntime_manager.go:541 +0x1d1
Apr 25 14:59:14 node0 kubelet[2447606]: k8s.io/kubernetes/pkg/kubelet/kuberuntime.(*kubeGenericRuntimeManager).SyncPod(0xc00084c180, 0xc001504800, 0xc00099d800, 0x558b483c77b0, 0x0, 0x0, 0xc000a65140, 0x0, 0x0, 0x0, ...)
Apr 25 14:59:14 node0 kubelet[2447606]:         /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/kuberuntime/kuberuntime_manager.go:726 +0x7e
Apr 25 14:59:14 node0 kubelet[2447606]: k8s.io/kubernetes/pkg/kubelet.(*Kubelet).syncPod(0xc0007e8a80, 0x558b4603a1f0, 0xc0013a30c0, 0x0, 0xc001504800, 0x0, 0xc00099d800, 0x0, 0x0)
Apr 25 14:59:14 node0 kubelet[2447606]:         /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet.go:1729 +0xcf2
Apr 25 14:59:14 node0 kubelet[2447606]: k8s.io/kubernetes/pkg/kubelet.(*podWorkers).managePodLoop.func1(0xc001ea7ee0, 0xc00077c990, 0xc001504800, 0xc001ea7db8, 0x0, 0x0)
Apr 25 14:59:14 node0 kubelet[2447606]:         /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/pod_workers.go:733 +0x343
Apr 25 14:59:14 node0 kubelet[2447606]: k8s.io/kubernetes/pkg/kubelet.(*podWorkers).managePodLoop(0xc00077c990, 0xc001028300)
Apr 25 14:59:14 node0 kubelet[2447606]:         /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/pod_workers.go:738 +0x36b
Apr 25 14:59:14 node0 kubelet[2447606]: k8s.io/kubernetes/pkg/kubelet.(*podWorkers).UpdatePod.func1(0xc00077c990, 0xc001028300)
Apr 25 14:59:14 node0 kubelet[2447606]:         /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/pod_workers.go:618 +0x6f
Apr 25 14:59:14 node0 kubelet[2447606]: created by k8s.io/kubernetes/pkg/kubelet.(*podWorkers).UpdatePod
Apr 25 14:59:14 node0 kubelet[2447606]:         /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/pod_workers.go:616 +0xe58`

### What did you expect to happen?

We expect that in case of such failures K8s retries the failing pod created operation to be able to start the Pod. It should not be stuck in a state which requires manual intervention.

### How can we reproduce it (as minimally and precisely as possible)?

You can create two or more pods before the Calico is started, and then start the Calico.

### Anything else we need to know?

The code that I found problematic is:	
https://github.com/kubernetes/kubernetes/blob/632ed300f2c34f6d6d15ca4cef3d3c7073412212/pkg/kubelet/kuberuntime/kuberuntime_manager.go#L512-L515
`sandboxStatus.Network.Ip` may occur a null pointer exception
We can make the following modifications:
`if !kubecontainer.IsHostNetworkPod(pod) && (sandboxStatus.Network == nil || sandboxStatus.Network.Ip == "") {
		klog.V(2).InfoS("Sandbox for pod has no IP address. Need to start a new one", "pod", klog.KObj(pod))
		return true, sandboxStatus.Metadata.Attempt + 1, sandboxStatus.Id
	}`

### Kubernetes version

<details>

```console
$ kubectl version 
# paste output here
```
kubelet version :v1.22.1
</details>

### Cloud provider

<details>

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by rphillips** [comment]
The proposed change looks reasonable. @H-W-WXL would you like to create a PR?

**Comment by H-W-WXL** [comment]
Okay, I'll revise it as soon as possible.

**Comment by wangyysde** [comment]
@H-W-WXL Are you preparing a PR for this bug? I am want to fix this bug if you are not working for it.

**Comment by H-W-WXL** [comment]
> @H-W-WXL Are you preparing a PR for this bug? I am want to fix this bug if you are not working for it.

I've submitted PR ,but there are some conflicts in this PR. If the PR cannot be merged, you could modify it.

**Comment by H-W-WXL** [comment]
> @H-W-WXL Are you preparing a PR for this bug? I am want to fix this bug if you are not working for it.
if this PR(https://github.com/kubernetes/kubernetes/pull/109042/files) has solved this problem? and it has been merged two days ago.if this problem has been solved ,I,ll close this issue and the PR I've submitted.

**Comment by wangyysde** [comment]
Yes , I think that this problem has be fixed by #109042.
